### PR TITLE
fix(Drawer): default props not working

### LINKED ISSUE #39711: Drawer组件无法关闭
### Reproduction link

[https://4x.ant.design/components/drawer-cn/#API](https://4x.ant.design/components/drawer-cn/#API)

### Steps to reproduce

1、引入Drawer组件
2、设置open属性和控制open属性的state状态
3、改变state为true，Drawer组件显示
4、改变state为false，也就是open属性为false，Drawer组件无法隐藏

```js
const [open, setOpen] = useState(false);
<Drawer
        title='选择'
        placement='right'
        onClose={onClose}
        open={open}
      >
        <p>Some contents...</p>
        <p>Some contents...</p>
        <p>Some contents...</p>
      </Drawer>
```

### What is expected?

open属性改变为false时，Drawer可以正常关闭

### What is actually happening?

open属性改变为false时，Drawer无法关闭

| Environment | Info |
| --- | --- |
| antd | 4.24.5 |
| React | 17 |
| System | macOS Monterey 12.6 |
| Browser | Chrome  108.0.5359.124（正式版本） (arm64) |

### LINKED ISSUE #39777: Drawer component is broken in 5.1.0
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/basic-antd-5-1-0-forked-9t2vtu?file=/demo.tsx)

### Steps to reproduce

https://codesandbox.io/s/basic-antd-5-1-0-forked-9t2vtu?file=/demo.tsx

### What is expected?

Drawer expanding with whole space

### What is actually happening?

See the sandbox sample

| Environment | Info |
| --- | --- |
| antd | 5.1.0 |
| React | 18 |
| System | chrome and Edge |
| Browser | chrome and Edge |

---
https://codesandbox.io/s/basic-antd-5-1-0-forked-9t2vtu?file=/demo.tsx

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/39711
close https://github.com/ant-design/ant-design/issues/39777

### 💡 Background and solution

具体改动见：https://github.com/react-component/drawer/pull/370

`<Drawer open={undefined} placement={undefined} />` 会覆盖 default props，不等效于 `<Drawer />`。

虽然 semver 会自动更新，但 dist 文件还是有问题，semver 也有可能安装到 6.1.0。这里强制升级上来。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix Drawer default props not working when `placement` `open` `width` are `undefined`.       |
| 🇨🇳 Chinese |  修复 Drawer `placement` `open` `width` 等参数为 undefined 时默认值不生效的问题。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: scripts/post-script.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const DEPRECIATED_VERSION = {
   '4.24.0': ['https://github.com/ant-design/ant-design/issues/38371'],
   '5.0.4': ['https://github.com/ant-design/ant-design/issues/39284'],
   '5.0.6': [],
```
Comment (afc163) [inline scripts/post-script.js:27]: 5.0.6 这里缺 issue 链接。

**Review by li-jia-nan — APPROVED** [review]

File: scripts/post-script.js (hunk 1 above)
Comment (MadCcc) [reply scripts/post-script.js:27]: 我记得是没有 issue 相关，应该是 rc-tour 错版的问题，打个注释吧

Comment (MadCcc) [inline scripts/post-script.js:27]:
```suggestion
  '5.0.6': [], // Issue with @rc-components/tour v1.0.3
```

**Comment by afc163** [comment]
图片

间隙性异常。

File: scripts/post-script.js [hunk 2]
```diff
@@ -24,7 +24,8 @@ const DEPRECIATED_VERSION = {
   ],
   '4.24.0': ['https://github.com/ant-design/ant-design/issues/38371'],
   '5.0.4': ['https://github.com/ant-design/ant-design/issues/39284'],
-  '5.0.6': [],
+  '5.0.6': [], // Issue with @rc-components/tour v1.0.3
```
Comment (afc163) [inline scripts/post-script.js:27]: 可以建一个，说清楚问题。

Comment (MadCcc) [inline scripts/post-script.js:27]:
```suggestion
  '5.0.6': ['https://github.com/ant-design/ant-design/issues/39807'],
```

**Pull request merged**